include/ginkgo/core/base/executor.hpp

```diff
@@ -1619,6 +1626,12 @@ class HipExecutor : public detail::ExecutorBase<HipExecutor>,
     }
 
 protected:
+#if (GINKGO_HIP_PLATFORM_NVCC == 1)
+    using device_type = NvidiaDevice;
```

Comment:
Maybe so that there is no ambiguity with DPC++ this can be called `device_class` or something like this?